src/entries/popup/pages/home/TokenDetails/TokenDetails.tsx

```diff
@@ -286,6 +289,17 @@ function MoreOptions({
   const { pinnedAssets, removedPinnedAsset, addPinnedAsset } =
     usePinnedAssetStore();
 
+  const { selectedToken, setSelectedToken } = useSelectedTokenStore();
+
+  const resetSelectedToken = useCallback(() => {
+    if (selectedToken) setSelectedToken();
+  }, [setSelectedToken, selectedToken]);
+
+  useEffect(() => {
+    // When component unmounts reset the selectedToken
+    return resetSelectedToken;
+  }, [resetSelectedToken]);
```

Comment:
Let's say user navigates from this page to other pages like swap or bridge we would need to reset the `selectedToken` state back to `undefined`, otherwise we would still have an active `selectedToken` state through out the app.